gammapy/datasets/utils.py

```diff
@@ -203,3 +214,165 @@ def split_dataset(dataset, width, margin, split_template_models=True):
                 d.models = dataset.models
             datasets.append(d)
     return datasets
+
+
+def create_map_dataset_from_dl4_irfs(obs, geom=None, energy_axis_true=None, name=None):
+    """Create a map dataset from an observation containing DL4 IRFs
+
+
+    Parameters
+    ----------
+    obs : `~gammapy.data.Observation`
+        Observation containing DL4 IRFs
+    geom : `~gammapy.maps.WcsGeom`, optional
+        Output dataset maps geometry. The default is None, and it is derived from IRFS
+    energy_axis_true : `~gammapy.maps.MapAxis`, optional
+        True energy axis used for IRF maps. The default is None, and it is derived from IRFS
+    name : str, optional
+        Dataset name. The default is None, and the name is randomly generated.
+
+    Returns
+    -------
+    dataset : `~gammapy.datasets.MapDataset`
+        Map dataset.
+
+    """
+    from gammapy.makers import MapDatasetMaker
+    from gammapy.datasets import MapDataset
+
+    # define target geom
+    if geom is None:
+        geom_image = obs.aeff.geom.to_image()
+        geom = geom_image.to_cube([obs.edisp.edisp_map.geom.axes["energy"]])
+
+    energy_axis = geom.axes["energy"]
+
+    if energy_axis_true is None:
+        # energy_axis_true = obs.edisp.edisp_map.geom.axes["energy_true"]
+        energy_axis_true = obs.aeff.geom.axes["energy_true"]
+
+    # ensure that DL4 IRFs have the axes
+    rad_axis = obs.psf.psf_map.geom.axes["rad"]
+    geom_psf = obs.psf.psf_map.geom.to_image().to_cube([rad_axis, energy_axis_true])
+    geom_edisp = obs.edisp.edisp_map.geom.to_image().to_cube(
+        [energy_axis, energy_axis_true]
+    )
+    geom_exposure = geom.to_image().to_cube([energy_axis_true])
+
+    # create dataset and run data reduction / irfs interpolation
+    dataset = MapDataset.from_geoms(
+        geom,
+        geom_exposure=geom_exposure,
+        geom_psf=geom_psf,
+        geom_edisp=geom_edisp,
+        name=name,
+    )
```

Comment:
@QRemy, did you figure out why sometimes having the IRFs in the original geom in HAWC analyses (so only 1 R.A. bin) caused issues with some of the estimators?